### DISCUSSION TITLE
Nginx seem stuck at the fresh install

### DISCUSSION THREAD

**Discussion by Railline** [ZoeyVid/NPMplus#484] [Q&A] [answered]
Hi, i got a little problem with NPM-Plus i want to change my old docker image form baudneo to this one more update, but on fresh install before i do any configuration i can't access to the admin interface or any port (i don't have firewall)

when i go to my log i see that
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/6c2d7b17-b746-455a-912c-23d4807bb0c7)
(i have wait more that 3 second)

I use it on unraid 6.12.4
i have try bridge and host
i have change the IPv4 Binding and NPM Ipv4 binding ( the binding for the admin interface stay in 127.0.0.1)
can someone help me i feel lost

**Comment by Zoey2936** [comment]
Please send your compose file

**Reply by Railline** [reply]
it's not really a compose, it is but on xml

```xml
<?xml version="1.0"?>
<Container version="2">
  <Name>NPM-Plus</Name>
  <Repository>zoeyvid/npmplus:latest</Repository>
  <Registry/>
  <Network>bridge</Network>
  <MyIP/>
  <Shell>sh</Shell>
  <Privileged>true</Privileged>
  <Support/>
  <Project/>
  <Overview/>
  <Category>Cloud: Security: Tools: Network:Web Network:Proxy Network:Management Status:Stable</Category>
  <WebUI>http://[IP]:[PORT:81]</WebUI>
  <TemplateURL/>
  <Icon>https://raw.githubusercontent.com/ItJustFox/unraidtemplate/main/Nginx-Proxy-Manager-CrowdSec/NginxCrowdsec.png</Icon>
  <ExtraParams>--hostname=NPMPlus</ExtraParams>
  <PostArgs/>
  <CPUset/>
  <DateInstalled>1699783269</DateInstalled>
  <DonateText/>
  <DonateLink/>
  <Requires/>
  <Config Name="WebUI" Target="81" Default="" Mode="tcp" Description="" Type="Port" Display="always" Required="false" Mask="false">81</Config>
  <Config Name="HTTP Port" Target="80" Default="" Mode="tcp" Description="" Type="Port" Display="always" Required="false" Mask="false">4480</Config>
  <Config Name="HTTPS Port" Target="443" Default="" Mode="tcp" Description="" Type="Port" Display="always" Required="false" Mask="false">44443</Config>
  <Config Name="TimeZone" Target="TZ" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">Europe/Paris</Config>
  <Config Name="NIBEP" Target="NIBEP" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">48694</Config>
  <Config Name="NPM_PORT" Target="NPM_PORT" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">81</Config>
  <Config Name="IPV4_BINDING" Target="IPV4_BINDING" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">0.0.0.0</Config>
  <Config Name="NPM_IPV4_BINDING" Target="NPM_IPV4_BINDING" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">0.0.0.0</Config>
  <Config Name="IPV6_BINDING" Target="IPV6_BINDING" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">[::1]</Config>
  <Config Name="NPM_IPV6_BINDING" Target="NPM_IPV6_BINDING" Default="" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">[::1]</Config>
  <Config Name="DISABLE_IPV6" Target="DISABLE_IPV6" Default="false|true" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">true</Config>
  <Config Name="NPM_DISABLE_IPV6" Target="NPM_DISABLE_IPV6" Default="false|true" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">true</Config>
  <Config Name="NPM_LISTEN_LOCALHOST" Target="NPM_LISTEN_LOCALHOST" Default="true|false" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">true</Config>
  <Config Name="CLEAN" Target="CLEAN" Default="false|true" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">false</Config>
  <Config Name="FULLCLEAN" Target="FULLCLEAN" Default="false|true" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">false</Config>
  <Config Name="PHP81" Target="PHP81" Default="false|true" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">false</Config>
  <Config Name="PHP81_APKS" Target="PHP81_APKS" Default="" Mode="" Description="Add php extensions, see available packages here: https://pkgs.alpinelinux.org/packages?branch=v3.18&amp;amp;repo=community&amp;amp;arch=x86_64&amp;amp;name=php81-*, default none" Type="Variable" Display="always" Required="false" Mask="false"/>
  <Config Name="PHP82" Target="PHP82" Default="false|true" Mode="" Description="" Type="Variable" Display="always" Required="false" Mask="false">false</Config>
  <Config Name="PHP82_APKS" Target="PHP82_APKS" Default="" Mode="" Description="Add php extensions, see available packages here: https://pkgs.alpinelinux.org/packages?branch=v3.18&amp;amp;repo=community&amp;amp;arch=x86_64&amp;amp;name=php82-*, default none" Type="Variable" Display="always" Required="false" Mask="false"/>
  <Config Name="Data" Target="/data" Default="" Mode="rw,shared" Description="" Type="Path" Display="always" Required="false" Mask="false">/mnt/user/appdata/NPM-Plus/Data</Config>
  <Config Name="PUID" Target="PUID" Default="" Mode="" Description="" Type="Variable" Display="advanced-hide" Required="false" Mask="false">1000</Config>
  <Config Name="PGID" Target="PGID" Default="" Mode="" Description="" Type="Variable" Display="advanced-hide" Required="false" Mask="false">1000</Config>
</Container>
```
this is my template
and my config just here
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/dc5257a5-d282-4d3b-93c0-fec8b880546c)
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/464ced3c-b6d1-43bb-9d5f-91eed0327cc5)
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/78302c37-6d3b-4ea5-b8cf-a1eb61b9e16c)

**Reply by Zoey2936** [reply]
is the health check green?

**Reply by Railline** [reply]
no it's a option to bybass apparmor and allow access to all peripheral

**Reply by Zoey2936** [reply]
so `docker inspect --format "{{json .State.Health.Status }}" npmplus` returns unhealthy?

**Reply by Railline** [reply]
infinity starting
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/715000a2-90f6-4a30-b9e8-e781b7c6ac50)

**Reply by Railline** [reply]
ah now it's healthy 
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/66b37d02-fc50-461a-8013-9cfd0db0dacb)
but log seem still freeze
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/2ca98b66-bc33-453c-8f3f-f01e463a497d)

**Reply by Zoey2936** [reply]
the log don't freeze, if there are no errors nginx launched (and everything should work)

**Reply by Railline** [reply]
after 30min the docker start but i still can't access to admin interface or port http and https

![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/be20b71f-990c-4358-80bf-66744bdb8182)

**Reply by Zoey2936** [reply]
if the healtcheck is healthy npm is running and the admin interface is reachable from inside the container, so the problem needs to be a firewall

**Reply by Railline** [reply]
ok i will use the real command to get log form docker 
i found the reason of the slow start of nginx
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/99d4b024-79e5-44af-90f4-84bb1b7d8234)
but it's strange because 
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/aa118619-a347-4a64-8022-3a96c81b50aa)

i have try to see if the docker can access to the admin interface but i get à 301
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/46865be8-54ce-42ab-a6d2-b195a1a3617a)

**Reply by Zoey2936** [reply]
Try curl -k https://127.0.0.1:81

**Reply by Railline** [reply]
hmmm i think it's work on local now but not outsite
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/007403b6-bf66-4c31-b3b2-e16e4a65c25a)

**Reply by Zoey2936** [reply]
As I said I think it is a firewall problem

**Reply by Railline** [reply, marked as answer]
i have find the solution it's not firewall, the NPM binding cannot be 0.0.0.0 but 192.168.1.213 (ip of docker) i don't why but the ip 0.0.0.0 is use by a interface,

i have access now thank u ^^

just one last question it's possible to use external mysql instead the sqllite i see a var DB_MYSQL_CA but nothing about a external database

**Reply by Zoey2936** [reply]
Yes, mariadb support is like from the upstream NPM, but undupported by me

**Reply by Railline** [reply]
okay thank for the information ^^

i got a problem when i have try to create my first proxy host
![image](https://github.com/ZoeyVid/NPMplus/assets/57727650/7810cc46-f8bf-414e-8c69-d0377c6d8166)
when i try to create another one it's make the docker crash

**Reply by Zoey2936** [reply]
can you please retry

**Reply by Railline** [reply]
Yeah I have try with another url but still same error, and crash after

**Reply by Zoey2936** [reply]
do you run npmplus in readonly mode?

**Reply by Railline** [reply]
No it's on RW

**Reply by Zoey2936** [reply]
can you maybe test it on another system with docker-compose? since I can't reproduce it

**Reply by ItJustFox** [reply]
Hello there (general kenoby) ! 
I'm [Fantucie](https://forums.unraid.net/profile/142238-fantucie/) unRAID community dev and also girlfriend of Railline o/

I took a look at the docker compose and the configuration he used, and boy did I want to shoot myself, it seems that the problem only came from his side and its configuration, since unRAID is a bit special, some of the things it put in were completely useless or incorrect, so I took the liberty of correcting them to get a working template that works without a single crash.

Since you don't have an unRAID server, has an community dev I can make the template available for everyone in the Community apps for an quick & fast download & use, if you doesn't mind, I would like to put your version instead of the two old NPM I was maintening on it.

**Reply by Zoey2936** [reply]
So it works now? And for unRAID template, I have no idea, just create one if you want.

**Reply by ItJustFox** [reply]
I might ask, do you know where is the logs of the proxyhost ? 
Has crowdsec require them to scan to see who to ban or not on the regular NPM there is an folder `/logs` with all the file of access log like `proxy-host-50_access.log` &  `proxy-host-50_error.log` (there is also an subfolder `letsencrypt` but lets forget it)

**Reply by Zoey2936** [reply]
logs folder is removed, no access logs, error-logs are logged to console. Crowdsec don't scans logs it is like a module of nginx. letsencrypt folder is moved to the data folder

**Reply by ItJustFox** [reply]
> logs folder is removed, no access logs, error-logs are logged to console. Crowdsec don't scans logs it is like a module of nginx. letsencrypt folder is moved to the data folder

Uh, crowdsec does read the nginx to be able to ban bad actor.
![image](https://github.com/ZoeyVid/NPMplus/assets/42358087/85a236f4-e7c2-45c6-b852-130cf11e8369)
(old path from NPM proxy-host log)
![image](https://github.com/ZoeyVid/NPMplus/assets/42358087/a5f25ad4-3296-431d-80ad-cb9317001137)
One of the random log that Crowdsec actually scan.
![image](https://github.com/ZoeyVid/NPMplus/assets/42358087/d8e864fe-25a0-4cc4-82be-8b0e52d8b7b0)

Crowdsec  does in fact scan the proxy-host-{id}_{access||error}.log to be able to ban or not poeple trying to pass with sql and or other.
You can see it too from the [Crowdsec Hub](https://app.crowdsec.net/hub/author/crowdsecurity/collections/nginx-proxy-manager).

**Reply by Zoey2936** [reply]
ahhh we talk about diffrent things. If you want to block blacklisted ips just follow the documentation in the readme. You want to add clients causing 403 errors to the blacklist, which can work but requires a diffrent collection, which doesn't exist at this point.

**Reply by ItJustFox** [reply]
> ahhh we talk about diffrent things. If you want to block blacklisted ips just follow the documentation in the readme. You want to add clients causing 403 errors to the blacklist, which can work but requires a diffrent collection, which doesn't exist at this point.

I might ask why you disabled these logs ?
Because without them using Crowdsec loose 95% of is use, even too the docker `GoAccess-NPM-Logs` wich use the exact same logs has Crowdsec do for the auto ban, is useless and can no more be used, isn't it possible to just bring back these logs inside an `/logs` or `/data/logs` because they do are very helpfull in all way ?

**Reply by Zoey2936** [reply]
The GDPR doesn't allow collecting personal data without a reason/permission. And IP-adresses are personal data

**Reply by Zoey2936** [reply]
(breaking laws is not helpfull)

**Reply by Zoey2936** [reply]
you can still check the docker log for errors / modsecurity alearts

**Reply by ItJustFox** [reply]
The currents log system from NPM is not suitable for GDPR because they do are services logs you can even check it [here](https://www.freecodecamp.org/news/how-to-stay-gdpr-compliant-with-access-logs/)
Specially in plus knowing that the data collected are in prevention of SQL and other attack, to block them the access to the service by banning them with Crowdsec log parser.

**Reply by Zoey2936** [reply]
Sorry but no, the European Court of Justice decided multiple times that even states themself aren't allowed to save IPs, even if just for a few weeks: https://curia.europa.eu/juris/document/document.jsf?text=&docid=265881&pageIndex=0&doclang=EN&mode=req&dir=&occ=first&part=1&cid=623107

**Reply by Zoey2936** [reply]
If you want you can create a PR, adding access log back when setting an env

**Reply by Zoey2936** [reply]
Hello, I've added access log back, but it uses a different log format then upstream NPM

**Reply by Railline** [reply]
That is nice but it's not really useful in this case crowdsec read log in specific format (defaut format of npm) to detect potential hack and ban them. That is not really a problem but if people want to use crowdsec with your docker, they will have to create a custom detector for crowdsec

**Reply by Zoey2936** [reply]
https://app.crowdsec.net/hub/author/ZoeyVid/collections/npmplus I've added a collection you can use

**Comment by ItJustFox** [comment]
**For the answer**
The Extra Parameters `--hostname=NPMPlus --name npm-plus` where not needed at all has we can run the docker on bridge and not host, this was the error causing the crash when creating SSL or start for some.

For the variable `IPV4_BINDING` & `NPM_IPV4_BINDING` instead of putting the host ip he gave to the docker, I just made `0.0.0.0` so NPM listen to what he can in tearm of IP.

**Reply by Railline** [reply]
i have put 0.0.0.0 but the interface admin was not reachable
